### DISCUSSION TITLE
PDF reports, other users access except admin, access control based on roles

### DISCUSSION THREAD

**Discussion by sworup** [openmsupply/msupply-dashboard-app#97] [Feature request]
### Description

There are features that the clients have recently been asking for more vocally:

- [ ] PDF reports: Currently we only have excel report
- [ ] User access to users other then admin: Most probably editor access
- [ ] Check which store currently logged in user has access to, and restrict access level to panel list etc based on that
- [ ] Make generated excel files' name more customisable. Specially clients have been asking for custom formattable dates at the end of filename.

> Since we have a list of features, so for now I have created a list page with features as pointer. Maybe we ought to create one page with all feature requests enlisted (kinda like an epic) and then add individual feature request as their own page. 
> This will do for now.

**Comment by ujwalSussol** [comment]
@sworup   Thanks,  @anildahalsussol   : think we need a group meeting to discuss all points.

**Reply by Gwalior-84** [reply]
> 
>     * [ ]  User access to users other then admin: Most probably editor access
> 

Can you expand / rewrite this one please @sworup ?  I don't quite understand it.

I don't seem to have edit permission for the Description, so you have to add the other items?

- We need to allow multiple templates so that each report can have its own template

**Reply by sworup** [reply]
> User access to users other then admin: Most probably editor access

Currently only admin users can see the excel report scheduler plugin, to add or edit to the plugin. There are talk to let clients to use the plugin interface. But if we want to do that now, we have to set the user as admin. Admin has access to everything. So the idea is to have some other user type for editing the plugin features and have them not be able to update panel queries, just be abled to edit the plugin. If we are to open access only, then it is easy, if we are to have restrictions there are some work involved.

**Reply by sworup** [reply]
> * We need to allow multiple templates so that each report can have its own template

this is definitely do-able. Need to dedicate time and do some planning. Specially since the sentiment is that we may be moving to Superset.
